src/openssl/ssl/context.cr

```diff
@@ -109,6 +109,90 @@ abstract class OpenSSL::SSL::Context
   end
 
   class Server < Context
+    protected def set_tlsext_servername_callback
+      cb = ->(ssl : LibSSL::SSL, cmd : LibC::Int, ctxptr : Void*) {
+        ctx = Box(Server).unbox(ctxptr)
+        if ctx.sni_fail_hard
+          ret = LibSSL::SSL_TLSEXT_ERR_OK
```

Comment:
I'm not sure I understand `sni_fail_hard`. If true then it will accept any hostname, which is weird.

Should OK and NOACK be reversed or should OK be something else?

Reply:
fixed in latest push. I wasn't initializing the contexts correctly throughout the spec, and that coupled with the sni_fail_ahrd issue let tests pass. sni_fail_hard, if set, should immediately throw an error to any SSL client connecting (see any of cloudflares sites).